docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,1013 @@
+
+Here is an attempted-exhaustive list of tasks the runtime is expected to carry out in each phase.
+
+initialization:
+  * accept new registrations of parachains and parathreads. Probably best to do this only once a session to avoid bitfield schemas shifting often (see details on availability bitfields below)
+  * determine scheduled parachains and parathreads for the upcoming block or blocks.
+  * determine validator assignments to scheduled paras for the upcoming block or blocks.
+  * remove blocks which have been pending availability for too long. this is tightly coupled with scheduling.
+  * handle the start of a new session - discard all candidates pending availability and note the upcoming validator set. 
+  * apply calls from upward messages - messages from parachains to the relay chain.
+
+during the block:
+  * Receive availability bitfields and move candidates from a pending availability to included state. See subsection below
+  * Receive new backed candidates to target for availability. See subsection below.
+  * Receive updates to configuration.
+
+process availability bitfields:
+  * We will accept an optional signed bitfield from each validator in each block.
+  * We need to check the signature and length of the bitfield for validity. 
+  * We will keep the most recent bitfield for each validator in the session. Each bit corresponds to a particular parachain candidate pending availability. Parachains are scheduled on every block, so we can assign a bit to each one of those. Parathreads are not scheduled on every block, and there may be a lot of them, so we probably don't want a dedicated bit in the bitfield for those. Since we want an upper bound on the number of parathreads we have scheduled or pending availability, a concept of "execution cores" used in scheduling (TODO) should be reusable here - have a dedicated bit in the bitfield for each core, and each core will be assigned to a different parathread over time.
```

Comment:
We're currently discussing on riot if "core" should actually refer to a bit in the bitfield:  Rob suggested that parachains should've their own bit so that they cannot be blocked on a bit being freed up.  In that case, parachains get scheduled onto bits for the entire session, but parachain validator groups rotate with which bits they vote.  We could've physical cores consisting of parachain validator groups and logical cores consisting of bits in the bit field.

Reply:
We're currently thinking parathreads and parachains behave opposite here, just so that only one thing is rotating in each case.